libraries/AP_InertialSensor/AP_InertialSensor_SCHA63T.cpp

```diff
@@ -0,0 +1,431 @@
+/*
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <utility>
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Math/AP_Math.h>
+
+#include "AP_InertialSensor_SCHA63T.h"
+#include <GCS_MAVLink/GCS.h>
+
+#if defined(HAL_GPIO_PIN_RESET)
+#include <hal.h>
+#endif
+
+#define S_SUM           0x0E
+#define R_S1            0x10
+#define A_S1            0x12
+#define C_S1            0x14
+#define C_S2            0x15
+#define G_FILT_DYN      0x16
+#define RESCTRL         0x18
+#define MODE            0x19
+#define A_FILT_DYN      0x1A
+#define SEL_BANK        0x1F
+#define SET_EOI         0x20
+
+#define ACCEL_BACKEND_SAMPLE_RATE   2000
+#define GYRO_BACKEND_SAMPLE_RATE    2000
+
+extern const AP_HAL::HAL& hal;
+
+#define CONSTANTS_ONE_G             (9.80665f)						// m/s^2
+#define int16_val(v, idx) ((int16_t)(((uint16_t)v[2*idx] << 8) | v[2*idx+1]))
+
+// CRC8-SAE J1850 (X8+X4+X3+X2+1) left move table
+const unsigned char	bTblCrc[256] = {
```

Comment:
could be static? or could be in AP_Math/crc.cpp?

Reply:
I fixed it as static. This CRC is specific for murata imu, so I want to leave it as it is.